src/Rootfs.jl

```diff
@@ -528,6 +528,59 @@ function expand_cxxstring_abis(ps::Vector{<:Platform})
     return Platform[p for p in Iterators.flatten(expand_cxxstring_abis.(ps))]
 end
 
+"""
+    preferred_libgfortran_version(platform::Platform, shard::CompilerShard)
+
+Return the libgfortran version preferred by the given platform or GCCBootstrap shard.
+"""
+function preferred_libgfortran_version(platform::Platform, shard::CompilerShard)
+    # Some input validation
+    if shard.name != "GCCBootstrap"
+        error("Shard must be `GCCBootstrap`")
+    end
+    if shard.target.arch != platform.arch || shard.target.libc != platform.libc
+        error("Incompatible platform and shard target")
+    end
+
+    if compiler_abi(platform).libgfortran_version != nothing
+        # Here we can't use `shard.target` because the shard always has the
+        # target as ABI-agnostic, thus we have also to ask for the platform.
+        return compiler_abi(platform).libgfortran_version
+    elseif shard.version < v"7"
```

Comment:
For these versions and the string ABIs below I followed [this table](https://github.com/JuliaPackaging/Yggdrasil/blob/df9066910be48af05fab2dfe480d0d4ec9547227/RootFS.md#compiler-shards), but note that this not up-to-date as we now have GCC 9, too.

Reply:
Hmmm, I am a little conflicted, because I feel that we are duplicating a lot of the logic inside of [`Pkg.BinaryPlatforms.gcc_version()`](https://github.com/JuliaLang/Pkg.jl/blob/6bdf5fcffb89b2b401d42cad2ab62611f63e54c9/src/BinaryPlatforms.jl#L99); just backwards.  I wonder if we can consolidate the two somehow.

In any case, this change is clearly needed.